src/ert/parsing/config_errors.py

```diff
@@ -10,63 +12,56 @@ class ConfigWarning(UserWarning):
 class ConfigValidationError(ValueError):
     def __init__(
         self,
-        errors: Union[str, List[Tuple[Optional[str], str]]],
+        errors: Union[str, List[Union[ErrorInfo, Tuple[Optional[str], str]]]],
         config_file: Optional[str] = None,
     ) -> None:
-        self.errors: List[Tuple[Optional[str], str]] = []
+        self.errors: List[ErrorInfo] = []
         if isinstance(errors, list):
             for err in errors:
                 if isinstance(err, str):
-                    self.errors.append((config_file, err))
+                    self.errors.append(ErrorInfo(message=err, filename=config_file))
```

Comment:
The if/else logic isn't quite right here. The type indicates that the list contains either ErrorInfo or a tuple. `isinstance(err, str)` should never be true.

Reply:
Oops! This dead case is now removed